pkg/controller/pravegacluster/upgrade.go

```diff
@@ -322,10 +322,78 @@ func (r *ReconcilePravegaCluster) syncControllerVersion(p *pravegav1alpha1.Prave
 	return true, nil
 }
 
+func (r *ReconcilePravegaCluster) syncStoreVersion(p *pravegav1alpha1.PravegaCluster) (synced bool, err error) {
+	//this is to check if the target image is above 07 and the current image is below 07
+	if p.Status.TargetVersion != "" && !util.IsVersionBelow07(p.Status.TargetVersion) && util.IsVersionBelow07(p.Status.CurrentVersion) {
+		return r.syncSegmentStoreVersionFrom06To07(p)
+	}
+	return r.syncSegmentStoreVersion(p)
+}
+
+func (r *ReconcilePravegaCluster) syncSegmentStoreVersionFrom06To07(p *pravegav1alpha1.PravegaCluster) (synced bool, err error) {
```

Comment:
Could we call this syncSegmentStoreVersionTo07()

Reply:
changed it